wolfcrypt/src/aes.c

```diff
@@ -1741,20 +1743,133 @@ static WARN_UNUSED_RESULT WC_INLINE word32 PreFetchTe(void)
         }
     }
     return x;
+#else
+    return 0;
+#endif
 }
 #else
 /* load sbox into cache by cache line stride */
 static WARN_UNUSED_RESULT WC_INLINE word32 PreFetchSBox(void)
 {
+#ifndef WOLFSSL_AES_TOUCH_LINES
     word32 x = 0;
     int i;
 
     for (i = 0; i < 256; i += WC_CACHE_LINE_SZ/4) {
         x &= Tsbox[i];
     }
     return x;
+#else
+    return 0;
+#endif
+}
+#endif
+#endif
+
+#ifdef WOLFSSL_AES_TOUCH_LINES
+#if WC_CACHE_LINE_SZ == 128
+    #define WC_CACHE_LINE_BITS 5
+    #define WC_CACHE_LINE_MASK_HI 0xe0
+    #define WC_CACHE_LINE_MASK_LO 0x1f
+    #define WC_CACHE_LINE_ADD     0x20
+#elif WC_CACHE_LINE_SZ == 64
+    #define WC_CACHE_LINE_BITS 4
+    #define WC_CACHE_LINE_MASK_HI 0xf0
+    #define WC_CACHE_LINE_MASK_LO 0x0f
+    #define WC_CACHE_LINE_ADD     0x10
+#elif WC_CACHE_LINE_SZ == 32
+    #define WC_CACHE_LINE_BITS 3
+    #define WC_CACHE_LINE_MASK_HI 0xf8
+    #define WC_CACHE_LINE_MASK_LO 0x07
+    #define WC_CACHE_LINE_ADD     0x08
+#elif WC_CACHE_LINE_SZ = 16
+    #define WC_CACHE_LINE_BITS 2
+    #define WC_CACHE_LINE_MASK_HI 0xfc
+    #define WC_CACHE_LINE_MASK_LO 0x03
+    #define WC_CACHE_LINE_ADD     0x04
+#endif
```

Comment:
Since WC_CACHE_LINE_SZ can be overridden at compile time there should probably be a #error or #warning case here for sizes not accounted for.

Reply:
Added #error